### PR TITLE
[DA-4439] DLW Dosage API

### PR DESCRIPTION
**Pull request by shahwaiz14**
## Description of changes/additions
- Added DLW API
- It also creates a new table that will be used to store the dlw information.

Here is a sample payload that will be sent from healthpro:

```
{
    "module": "3",
    "visitperiod": "Period1DLW",
    "batchid": "NPHDLW9172397",
    "participantweight": "56.38",
    "dose": "84",
    "calculateddose": "84.57",
    "dosetime": "2022-11-03T08:45:49Z"
}
```

## Tests
- ✅  unit tests

[DA-4439]: https://precisionmedicineinitiative.atlassian.net/browse/DA-4439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rdr_service/model/study_nph.py [hunk 1]
```diff
@@ -414,3 +414,24 @@ class ParticipantOpsDataElement(NphBase):
...
+    created = Column(UTCDateTime)
+    modified = Column(UTCDateTime)
+    ignore_flag = Column(TINYINT, default=0)
+    participant_id = Column(BigInteger, ForeignKey("participant.id"), nullable=False)
+    module = Column(Enum(ModuleTypes), nullable=False)
+    visit_period = Column(Enum(VisitPeriod), nullable=False)
+    batch_id = Column(String(32), nullable=False)
+    participant_weight = Column(String(32), nullable=False)
+    dose = Column(String(32), nullable=False)
+    calculated_dose = Column(String(32), nullable=False)
```
Comment (shahwaiz14) [inline rdr_service/model/study_nph.py:432]: These fields store values like `{... "participantweight": "56.38", "dose": "84", "calculateddose": "84.57",..}`. is it ok to store them as string?

File: rdr_service/api/nph_participant_biobank_order_api.py [hunk 2]
```diff
@@ -104,3 +104,19 @@ def patch(self, nph_participant_id, rdr_order_id):
...
+
+    @auth_required(RDR_AND_HEALTHPRO)
+    def post(self, nph_participant_id):
+        resource = self.get_request_json()
+        m = self._get_model_to_insert(resource, nph_participant_id)
+        result = self._do_insert(m)
+
+        log_api_request(log=request.log_record, model_obj=result)
+        self._archive_request_log()
+        return self._make_response(result)
```
Comment (shahwaiz14) [inline rdr_service/api/nph_participant_biobank_order_api.py:122]: I had `super().post(participant_id=nph_participant_id)` here, but for some reason the 2 lines in base_api's `post` mentioned below 👇  were causing issues.
```
if GAE_PROJECT == 'localhost':
                if ov_id:
                    onsite_id_verification_build(ov_id)
                else:
                    bq_participant_summary_update_task(participant_id)   <---
                    rebuild_participant_summary_resource(participant_id)  <---
```

File: rdr_service/model/study_nph.py (hunk 1 above)
Comment (dtharpeuno) [reply rdr_service/model/study_nph.py:432]: @shahwaiz14 casting them as strings is fine, its a pattern that we often do.

File: rdr_service/dao/study_nph_dao.py [hunk 3]
```diff
@@ -1397,3 +1398,56 @@ def update(self, obj: Incident) -> Incident:
...
+        if module not in valid_modules:
+            raise BadRequest(f'Invalid module provided. Module should be in {valid_modules}')
+
+        # Validate Visit Period
+        visit_period = resource.get('visitperiod', '').upper()
+        if visit_period not in valid_visit_periods:
+            raise BadRequest(f'Invalid visitPeriod provided. Visit Period should be in {valid_visit_periods}')
+
+    # pylint: disable=unused-argument
+    def from_client_json(self, resource, participant_id, client_id):
```
Comment (dtharpeuno) [inline rdr_service/dao/study_nph_dao.py:1436]: Can we try to pass this as an underscore so we don't have to use the pylint bypass?

`from_client_json(self, resource, participant_id, _):`

**Review by dtharpeuno — APPROVED** [review]

File: rdr_service/model/study_nph.py (hunk 1 above)
Comment (shahwaiz14) [reply rdr_service/model/study_nph.py:432]: Got it, thanks Darryl

File: rdr_service/api/nph_participant_biobank_order_api.py (hunk 2 above)
Comment (j-kanuch) [reply rdr_service/api/nph_participant_biobank_order_api.py:122]: How you have it is fine. Those lines are for PDR and are not necessary for this API.

**Review by j-kanuch — APPROVED** [review]
Looks good!

File: rdr_service/dao/study_nph_dao.py (hunk 3 above)
Comment (shahwaiz14) [reply rdr_service/dao/study_nph_dao.py:1436]: @dtharpeuno I think we do need to pass in the `client_id` explicitly here b/c of this
```
def _get_model_to_insert(self, resource, participant_id=None):
        # Children of participants accept a participant_id parameter to from_client_json; others don't.
        if participant_id:
            return self.dao.from_client_json(
                resource, participant_id=participant_id, client_id=app_util.get_oauth_id()
            )
        else:
            return self.dao.from_client_json(resource, client_id=app_util.get_oauth_id())
```

**Pull request merged**